code/modules/events/ghost_role/fugitive_event.dm

```diff
@@ -15,25 +17,25 @@
 
 /datum/round_event/ghost_role/fugitives/spawn_role()
 	var/list/possible_spawns = list()//Some xeno spawns are in some spots that will instantly kill the refugees, like atmos
-	for(var/turf/X in GLOB.xeno_spawn)
-		if(istype(X.loc, /area/station/maintenance))
-			possible_spawns += X
+	for(var/turf/spawn_turf in GLOB.xeno_spawn)
+		if(istype(get_area(spawn_turf), /area/station/maintenance) && is_safe_turf(spawn_turf))
+			possible_spawns += spawn_turf
 	if(!possible_spawns.len)
 		message_admins("No valid spawn locations found, aborting...")
 		return MAP_ERROR
 	var/turf/landing_turf = pick(possible_spawns)
 	var/list/possible_backstories = list()
 	var/list/candidates = get_candidates(ROLE_FUGITIVE, ROLE_FUGITIVE)
-	if(candidates.len >= 1) //solo refugees
-		if(prob(30))
-			possible_backstories.Add("waldo") //less common as it comes with magicks and is kind of immershun shattering
-		else //For accurate deadchat feedback
-			minimum_required = 4
-	if(candidates.len >= 4)//group refugees
-		possible_backstories.Add("prisoner", "cultist", "synth")
-	if(!possible_backstories.len)
+
+	if(!length(candidates))
 		return NOT_ENOUGH_PLAYERS
 
+	if(length(candidates) < TEAM_BACKSTORY_SIZE || prob(30 - (length(candidates) * 2))) //Solo backstories are always considered if a larger backstory cannot be filled out. Otherwise, it's a rare chance that gets rarer if more people sign up.
+		possible_backstories += list("waldo") //less common as it comes with magicks and is kind of immershun shattering
```

Comment:
honestly these should be defines instead of raw strings to prevent against typos